tock/hours/models.py

```diff
@@ -316,24 +316,35 @@ def calculate_target_hours(self):
             1. Target hours in a regular 40 hour work week
             2. Billable expectation * number of hours worked
         """
-        this_weeks_target = round(self.billable_expectation * self.calculate_utilization_denominator(), 0)
+        this_weeks_target = round(self.billable_expectation * self.calculate_hourly_utilization_denominator(), 0)
         return min(this_weeks_target, self.max_target_hours())
 
     def calculate_total_weekly_allocation(self):
         """
         Loops through related time card objects and sums total
         weekly allocation percentage
         """
-        timecardobjs = self.timecardobjects.all()
+        timecardobjs = self.timecardobjects.filter(project__accounting_code__billable=True)
         total_weekly_allocation = 0.00
         for tco in timecardobjs:
             total_weekly_allocation += float(tco.project_allocation)
         return total_weekly_allocation
 
     def calculate_utilization(self):
+        """
+        Calculate this timecard's utilization using the sum of weekly allocation
+        and the billable hours spent compared with the target for billable hours
+
+        Return None for timecards with neither billable allocation nor target hours
+        """
         if self.target_hours == 0:
-            return None
-        return self.billable_hours / self.target_hours
+            if self.total_weekly_allocation == 0.00:
+                return None
+            else:
+                return self.total_weekly_allocation
+        else:
+            hourly_utilization = float(self.billable_hours / self.target_hours)
```

Comment:
Thought: For a user not on a weekly billing project, `self.target_hours` can be relied upon to either be a [user's billable expectation multiplied by the max hours in a work week setting](https://github.com/18F/tock/blob/utilization-timecard/tock/hours/models.py#L349-L350) or 0 (if the user [has not yet submitted their timecard](https://github.com/18F/tock/commit/1b6c1475c5ed63ca7050cb7971b0f3c1e98203b4)). 

For users with a weekly billing project, this falls apart. `self.target_hours` can now be any value from 0 to expectation * max hours in a work week (generally 32?). This is the case where a user has a weekly billing project on their time card and tocks a few hours to a billable hourly project (e.g., picks up a help wanted request). For example:

* 1 weekly project at 100%
* 1 billable project at 2 hours
* `self.target_hours` = 2 hours (2 billable hours + 0 non-billable hours)

In this case, you'd end up with a utilization of 200%: 100% weekly + 100% hourly (2 billable hours / 2 total hours).

The appropriate change to account for this *might* be to adjust `calculate_target_hours` to return either 0 if the user has not submitted a timecard or `max_target_hours` if they have ([it's not totally clear to me why this is not already the approach](https://github.com/18F/tock/pull/1038)). In that case, the example above would result in a combined utilization of 106.25% (2 / 32 = .0625 hourly utilization + 1.0 weekly utilization).

Reply:
Thaaat's a real good point.  I think billable hours divided by (max_target_hours - OOO time) could work?

Reply:
Aha, OOO time is why `calculate_target_hours` doesn't just return 0 or `max_target_hours`! My explanation in my first paragraph was wrong then, `self.target_hours` can be other values in the case where a user tocked OOO or other time marked as `exclude_from_billability`.